exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -491,259 +648,563 @@ func (e *Exchange) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Sub
+				return nil, err
+			}
+			return s.DeriveSubmitResponse(response.ID)
+		default:
+			return nil, fmt.Errorf("%w: %v", order.ErrUnsupportedOrderType, s.Type)
+		}
+	case asset.Futures:
+		side := "BUY"
+		if s.Side.IsShort() {
+			side = "SELL"
+		}
+		var stpMode string
+		switch s.TimeInForce {
+		case order.PostOnly:
+			stpMode = "EXPIRE_MAKER"
+		case order.GoodTillCancel:
+			stpMode = "EXPIRE_TAKER"
+		}
+		switch s.Type {
+		case order.Market, order.Limit, order.LimitMaker:
+		default:
+			return nil, fmt.Errorf("%w: %v", order.ErrUnsupportedOrderType, s.Type)
+		}
+		var positionSide order.Side
+		if s.MarginType != margin.Unset {
+			positionSide = s.Side.Position()
+		}
+		response, err := e.PlaceFuturesOrder(ctx, &FuturesOrderRequest{
+			ClientOrderID:           s.ClientOrderID,
+			Side:                    side,
+			MarginMode:              marginMode(s.MarginType),
+			PositionSide:            positionSide,
+			Symbol:                  s.Pair.String(),
+			OrderType:               OrderType(s.Type),
+			ReduceOnly:              s.ReduceOnly,
+			TimeInForce:             TimeInForce(s.TimeInForce),
+			Price:                   s.Price,
+			Size:                    s.Amount,
+			SelfTradePreventionMode: stpMode,
+		})
+		if err != nil {
+			return nil, err
+		}
+		return s.DeriveSubmitResponse(response.OrderID)
+	default:
+		return nil, fmt.Errorf("%w: %q", asset.ErrNotSupported, s.AssetType)
 	}
-	return s.DeriveSubmitResponse(strconv.FormatInt(response.OrderNumber, 10))
 }
 
 // ModifyOrder modifies an existing order
 func (e *Exchange) ModifyOrder(ctx context.Context, action *order.Modify) (*order.ModifyResponse, error) {
 	if err := action.Validate(); err != nil {
 		return nil, err
 	}
-
-	oID, err := strconv.ParseInt(action.OrderID, 10, 64)
-	if err != nil {
-		return nil, err
-	}
-
-	resp, err := e.MoveOrder(ctx,
-		oID,
-		action.Price,
-		action.Amount,
-		action.TimeInForce.Is(order.PostOnly),
-		action.TimeInForce.Is(order.ImmediateOrCancel))
-	if err != nil {
-		return nil, err
-	}
-
-	modResp, err := action.DeriveModifyResponse()
-	if err != nil {
-		return nil, err
+	if action.AssetType != asset.Spot {
+		return nil, fmt.Errorf("%w: %q", asset.ErrNotSupported, action.AssetType)
+	}
+	switch action.Type {
+	case order.Market, order.Limit, order.LimitMaker:
+		resp, err := e.CancelReplaceOrder(ctx, &CancelReplaceOrderRequest{
+			OrderID:           action.OrderID,
+			ClientOrderID:     action.ClientOrderID,
+			Price:             action.Price,
+			Quantity:          action.Amount,
+			AmendedType:       action.Type.String(),
+			TimeInForce:       TimeInForce(action.TimeInForce),
+			SlippageTolerance: 0,
+		})
+		if err != nil {
+			return nil, err
+		}
+		modResp, err := action.DeriveModifyResponse()
+		if err != nil {
+			return nil, err
+		}
+		modResp.OrderID = resp.ID
+		return modResp, nil
+	case order.Stop, order.StopLimit:
+		oResp, err := e.CancelReplaceSmartOrder(ctx, &CancelReplaceSmartOrderRequest{
```

Comment:
Has no test coverage
If it does, please stop using `require.NoError` and use `assert.NoError` for multi-case tests